pynxtools/dataconverter/readers/ellips/reader.py

```diff
@@ -24,10 +24,78 @@
+        'INSTRUMENT[instrument]/BEAM_PATH[beam_path]/focussing_probes/angular_spread',
+    'angular_spread/@units':
+        'INSTRUMENT[instrument]/BEAM_PATH[beam_path]/focussing_probes/angular_spread/@units',
+    'atom_types': 'SAMPLE[sample]/atom_types',
+    'backside_roughness': 'SAMPLE[sample]/backside_roughness',
+    'calibration_status': 'INSTRUMENT[instrument]/calibration_status',
+    'chemical_formula': 'SAMPLE[sample]/chemical_formula',
+    'column_names': 'data_collection/column_names',
+    'company': 'INSTRUMENT[instrument]/company',
+    'count_time': 'INSTRUMENT[instrument]/BEAM_PATH[beam_path]/DETECTOR[detector]/count_time',
+    'count_time/@units':
+        'INSTRUMENT[instrument]/BEAM_PATH[beam_path]/DETECTOR[detector]/count_time/@units',
+    'data_correction':
+        'INSTRUMENT[instrument]/BEAM_PATH[beam_path]/focussing_probes/data_correction',
+    'data_error': 'data_collection/data_error',
+    'data_identifier': 'data_collection/data_identifier',
+    'data_software/@url': 'data_collection/data_software/@url',
+    'data_software/program': 'data_collection/data_software/program',
+    'data_software/version': 'data_collection/data_software/version',
+    'data_type': 'data_collection/data_type',
+    'depends_on': 'INSTRUMENT[instrument]/BEAM_PATH[beam_path]/depends_on',
+    'depolarization': 'derived_parameters/depolarization',
+    'detector_type':
+        'INSTRUMENT[instrument]/BEAM_PATH[beam_path]/DETECTOR[detector]/detector_type',
+    'ellipsometer_type': 'INSTRUMENT[instrument]/ellipsometer_type',
+    'layer_structure': 'SAMPLE[sample]/layer_structure',
+    'light_source': 'INSTRUMENT[instrument]/BEAM_PATH[beam_path]/light_source',
+    'medium': 'INSTRUMENT[instrument]/sample_stage/environment_conditions/medium',
+    'measured_data': 'data_collection/measured_data',
+    'model': 'INSTRUMENT[instrument]/model',
+    'model/@version': 'INSTRUMENT[instrument]/model/@version',
+    'real_time': 'INSTRUMENT[instrument]/BEAM_PATH[beam_path]/DETECTOR[detector]/real_time',
+    'revolutions': 'INSTRUMENT[instrument]/BEAM_PATH[beam_path]/rotating_element/revolutions',
+    'rotating_element_type': 'INSTRUMENT[instrument]/rotating_element_type',
+    'sample_history': 'SAMPLE[sample]/sample_history',
+    'sample_name': 'SAMPLE[sample]/sample_name',
+    'sample_type': 'SAMPLE[sample]/sample_type',
+    'software/@url': 'INSTRUMENT[instrument]/software/@url',
+    'software/program': 'INSTRUMENT[instrument]/software/program',
+    'software/version': 'INSTRUMENT[instrument]/software/version',
+    'stage_type': 'INSTRUMENT[instrument]/sample_stage/stage_type',
+    'substrate': 'SAMPLE[sample]/substrate',
+}
+
+CONFIG_KEYS = [
+    'colnames',
+    'derived_parameter_type',
+    'err-var',
+    'filename',
+    'parameters',
+    'plot_name',
+    'sep',
+    'skip',
+    'spectrum_type',
+    'spectrum_unit'
+]
+
+REPLACE_NESTED = {
+    'INSTRUMENT[instrument]': 'INSTRUMENT[instrument]',
```

Comment:
This seems to do nothing here.

Reply:
Right. This is a placeholder (for an empty list I got an error message...). After updating the ELN schema, some keys to be replaced will be added here.

Reply:
Ok, if the empty list produces an error we should fix this in the `flatten_and_replace` function as it should also be possible to leave this empty.

I just looked at the code there and it just uses the `items()` method on this replace nested dict, which should be fine for an empty list. So I don't understand where the error was coming from. Could you post your error message you get?